nwplus/nwhacks2019
# Add loading animation and error handling

:tickets: **Ticket(s)**: Closes rest of #159 

---

## :construction_worker: Changes

After recaptcha token is generated and our cloud function is called, the submit button now becomes a "Submitting..." button. Also if our function fails for some reason, it will show an error message on the UI and put the submit button back into it's normal state.

+ also added unrelated change 277d2c7450b9bb191e7e5a863296d5006c765108

## :thought_balloon: Notes

This doesn't handle failing to contact the reCAPTCHA servers (i.e. if they're down or in the more likely case that the user is disconnected from the internet). In this case, the button never goes into the "Submitting..." state.

## :flashlight: Testing Instructions

**[Here's a demo gif where I follow the below steps.](https://i.imgur.com/cpb44bQ.gif)**
1. Run `make web` and try to submit the application - it will fail because the submit function isn't being emulated (i.e. fails to response). You should see the "Submitting..." message followed by the error message on the UI. Try clicking on it multiple times to see that it still fails.
2. Run `make functions` in a separate terminal and now try the submit button. It should succeed.
